allennlp/training/trainer.py

```diff
@@ -631,6 +631,9 @@ def train(self) -> None:
 
         logger.info("Beginning training.")
 
+        train_metrics = None
+        val_metrics = None
+        epochs_trained = 0
```

Comment:
why do you track this explicitly? you have `epoch_counter` and the final value of `epoch`